joomla/joomla-cms
# [4.4] Move to ESM

Pull Request for Issue # .

### Summary of Changes

- Move tools and source JS to ESM
- ES6 files have now the extension `.mjs` (used to be `.es6.js`)
- Web Components also have `.mjs` extension (used to be `.w-c.es6.js`)
- The `build/media_source/system/js/joomla-core-loader.mjs` file now includes the required css and the related `build/media_source/system/scss/joomla-core-loader.scss` is deleted

### Testing Instructions

You need Git and NPM to test this

Pull the `4.4-dev` branch and run:
- `npm install`
- `npm run gzip`
- `npm run versioning`
- Store the media folder somewhere outside of the `joomla-cms` folder

Pull this PR `gh pr checkout 43779` and run:
- `npm install`
- `npm run gzip`
- `npm run versioning`
- Store the media folder somewhere outside of the `joomla-cms` folder


Compare the two folders. The only difference should be the missing files: `media/system/css/joomla-core-loader.css`, `media/system/css/joomla-core-loader.min.css`, `media/system/css/joomla-core-loader.min.css.gz`

Check that basic backend functionality


### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Link to documentations
Please select:



@Fedik @richard67 could you please test this one (pretty hard for the average tester)?


@LadySolveig @bembelimen please DO NOT upmerge this, I will do a PR on the 5.1 instead
